fda/fdata_boxplot.py

```diff
@@ -0,0 +1,627 @@
+
+    It the data of the functional boxplot or surface boxplot of a FDataGrid object,
+    depending on the dimensions of the domain, 1 or 2 respectively.
+
+    It forces to both classes, Boxplot and SurfaceBoxplot to conain at least the median,
+    central and outlying envelopes and a colormap for their graphical representation,
+    obtained calling the plot method.
+
+    """
+    @abstractmethod
+    def __init__(self, factor=1.5):
+        if factor < 0:
+            raise ValueError(
+                "The number used to calculate the outlying envelope must be positive.")
+
+    @property
+    def fdatagrid(self):
+        pass
+
+    @property
+    def median(self):
+        pass
+
+    @property
+    def central_envelope(self):
+        pass
+
+    @property
+    def outlying_envelope(self):
+        pass
+
+    @property
+    def colormap(self):
+        pass
+
+    @colormap.setter
+    def colormap(self, value):
+        pass
+
+    @abstractmethod
+    def plot(self, fig=None, ax=None, nrows=None, ncols=None):
+        pass
+
+    @abstractmethod
+    def _repr_svg_(self):
+        pass
+
+
+class Boxplot(FDataBoxplot):
+    r"""Representation of the functional boxplot.
+
+    Class implementing the functionl boxplot which is an informative exploratory
+    tool for visualizing functional data, as well as its generalization, the
+    enhanced functional boxplot. Only supports 1 dimensional domain functional data.
+
+    Based on the center outward ordering induced by a :ref:`depth measure <depth-measures>`
+    for functional data, the descriptive statistics of a functional boxplot are: the
+    envelope of the 50% central region, the median curve,and the maximum non-outlying envelope.
+    In addition, outliers can be detected in a functional boxplot by the 1.5 times the 50%
+    central region empirical rule, analogous to the rule for classical boxplots.
+
+    Attributes:
+        fdatagrid (FDataGrid): Object containing the data.
+        median (array, (fdatagrid.ndim_image, nsample_points)): contains
+            the median/s.
+        central_envelope (array, (fdatagrid.ndim_image, 2, nsample_points)):
+            contains the central envelope/s.
+        outlying_envelope (array, (fdatagrid.ndim_image, 2, nsample_points)):
+            contains the outlying envelope/s.
+        colormap (matplotlib.colors.LinearSegmentedColormap): Colormap from
+            which the colors to represent the central regions are selected.
+        central_regions (array, (fdatagrid.ndim_image * ncentral_regions, 2,
+            nsample_points)): contains the central regions.
+        outliers (array, (fdatagrid.ndim_image, fdatagrid.nsamples)):
+            contains the outliers
+        barcol (string): Color of the envelopes and vertical lines.
+        outliercol (string): Color of the ouliers.
+        mediancol (string): Color of the median.
+        fullout (boolean): If true, the entire curve of the outlier samples
```

Comment:
In the example, by default the curves are shown, and setting this to True hides them. Thus, either the code or the documentation is wrong. I would also change the name for a more understandable one.

Reply:
I have changed it in order the documentation and implementation make sense. Also I changed the name to outliers_repr but I do not like it much, any suggestions?

Reply:
`show_full_outliers`?